packages/mgt-components/src/components/mgt-file-list/mgt-file-list.ts

```diff
@@ -0,0 +1,606 @@
+  }
+
+  /**
+   * Render the list of files.
+   *
+   * @protected
+   * @param {*} files
+   * @returns {TemplateResult}
+   * @memberof mgtFileList
+   */
+  protected renderFiles(): TemplateResult {
+    return html`
+      <div id="file-list-wrapper" class="file-list-wrapper">
+        <ul id="file-list" class="file-list">
+          ${repeat(
+            this.files,
+            f => f.id,
+            f => html`
+              <li class="file-item">
+                ${this.renderFile(f)}
+              </li>
+            `
+          )}
+        </ul>
+        ${this.pageIterator && this.pageIterator.hasNext ? this.renderMoreFileButton() : null}
+      </div>
+    `;
+  }
+
+  /**
+   * Render an individual file.
+   *
+   * @protected
+   * @returns {TemplateResult}
+   * @memberof mgtFileList
+   */
+  protected renderFile(file: MicrosoftGraph.DriveItem): TemplateResult {
+    const view = this.itemView;
+    return (
+      this.renderTemplate('file', { file }) ||
+      html`
+        <mgt-file .fileDetails=${file} .view=${view}></mgt-file>
+      `
+    );
+  }
+
+  /**
+   * Render the button when clicked will show more files.
+   *
+   * @protected
+   * @returns {TemplateResult}
+   * @memberof MgtFileList
+   */
+  protected renderMoreFileButton(): TemplateResult {
+    if (this.isLoadingState) {
+      return html`
+        <mgt-spinner></mgt-spinner>
+      `;
+    } else {
+      return html`<a id="show-more" class="show-more" @click=${() =>
+        this.renderNextPage()}><span>Show more items<span></a>`;
+    }
+  }
+
+  /**
+   * load state into the component.
+   *
+   * @protected
+   * @returns
+   * @memberof MgtFileList
+   */
+  protected async loadState() {
+    const provider = Providers.globalProvider;
+    if (!provider || provider.state === ProviderState.Loading) {
+      return;
+    }
+
+    if (provider.state === ProviderState.SignedOut) {
+      this.files = null;
+      return;
+    }
+    const graph = provider.graph.forComponent(this);
+    let files: MicrosoftGraph.DriveItem[];
+    let pageIterator: GraphPageIterator<MicrosoftGraph.DriveItem>;
+
+    const getFromMyDrive = !this.driveId && !this.siteId && !this.groupId && !this.userId;
+
+    if (
+      (this.driveId && (!this.itemId && !this.itemPath)) ||
+      (this.groupId && (!this.itemId && !this.itemPath)) ||
+      (this.siteId && (!this.itemId && !this.itemPath)) ||
+      (this.userId && (!this.insightType && (!this.itemId && !this.itemPath)))
+    ) {
+      this.files = null;
+    }
+
+    if (!this.files) {
+      if (this.fileListQuery && this.fileListQuery !== null) {
+        pageIterator = await getFilesByListQueryIterator(graph, this.fileListQuery, this.pageSize);
+        files = pageIterator.value;
+      } else if (this.fileQueries && this.fileQueries !== null) {
+        files = await getFilesByQueries(graph, this.fileQueries);
+      } else if (getFromMyDrive) {
+        if (this.itemId && this.itemId !== null) {
```

Comment:
This is a little redundant. If `this.itemId` evaluates as true, it is never going to be null.

Reply:
If you have item-id added as an attribute but not add any value, storybook hangs.

Reply:
I'm confused. What value does storybook set it to? My point is, if you are simply detecting truthiness of the attribute, the first check will do it. `this.whatever` will return true for any non-null value, so the `!== null` check is redundant.

Reply:
You can supply a component as this: `<mgt-file-list item-id="" />` and `itemId` exists but evaluates to `null`. `this.itemId` is true.

Reply:
Yeah javascript has a weird definition on null vs. undefined. And typeof null is an object. 🤣🙈🤷🏻‍♀️

Reply:
Fricken JS! Ok, ya there is definitely some weirdness here. It bothered me so I had to test a bit. Storybook aside, passing an empty attribute `item-id=""` causes the value to reflect as an empty string object `""`.

Because JS is weird, the evaluation of `(this.itemId && this.itemId !== null)` results as... also an empty string `""`; the value of the first evaluated object, not type-coerced to a bool as I expected.

So in the end, we are really just evaluating the empty string as truthy/falsy followed by a literal null check.

Imo, this is nonobvious behavior and I think it needs some sort of commenting or be tweaked a bit. Using double negation `!!` would make it more clear to me because I can tell that we are explicitly evaluating the truthiness of the attribute `if (!!this.itemId) {` 

Thoughts?